### PR TITLE
Redesign login API for web-based auth

### PR DESCRIPTION
**Pull request by ollpu**

### CODE REVIEW

File: api/mock_server/app.py [hunk 1]
```diff
@@ -38,14 +38,16 @@
 
 
 def login_post():
-    token = state.login(connexion.request.json)
+    token = state.login()
     print(f"got token: {token}")
     if token is not None:
```
Comment (kluopaja) [inline api/mock_server/app.py:43]: Can be removed.

File: api/mock_server/app.py [hunk 2]
```diff
@@ -38,14 +38,16 @@
...
-        return ({"X-Auth-Token": token}, 200)
+        return ({"X-Auth-Token": token, "authentication_url": "http://todo"}, 200)
 
     return ({"message": "Invalid username/password",
              "code": "invalid_credentials"}, 401)
```
Comment (kluopaja) [inline api/mock_server/app.py:47]: Also this

**Review by kluopaja — APPROVED** [review]

File: api/mock_server/app.py [hunk 3]
```diff
@@ -36,24 +36,37 @@
...
+        },
+        200
+    )
+
+@app.route('/authorize-login')
+def authorize_login_post():
+    token = connexion.request.args.get("token")
+    fail = connexion.request.args.get("fail") is not None
+    state.authorize_login(token, fail)
+    return "", 204
```
Comment (kluopaja) [inline api/mock_server/app.py:60]: What is the point of `fail` here?

File: tests/integration/login.rs [hunk 4]
```diff
@@ -8,11 +8,15 @@ fn login_is_successful_with_correct_username_and_password() {
...
 }
 
-#[distributed_slice(TESTS)]
-fn login_fails_with_incorrect_username() {
-    let assert = unsuccessful_login();
-    assert.failure();
-}
+// #[distributed_slice(TESTS)]
+// fn login_fails_with_incorrect_username() {
+//     let assert = unsuccessful_login();
```
Comment (H4m5t3r) [inline tests/integration/login.rs:13]: Can be removed? (Now unnecessary function call?)

**Review by H4m5t3r — APPROVED** [review]

File: api/mock_server/app.py (hunk 3 above)
Comment (ollpu) [reply api/mock_server/app.py:60]: I suppose to allow making an invalid token for testing purposes. Maybe it should be made easier to do after having initially logged in successfully?

**Pull request merged**